src/AuthFlowFactoryTrait.php

```diff
@@ -12,13 +12,15 @@
 
 use League\OAuth2\Server\AuthorizationServer;
 use Psr\Container\ContainerInterface;
-use Psr\Http\Message\ResponseInterface;
-
 use function sprintf;
 
-class OAuth2MiddlewareFactory
+/**
+ * Provides helper methods for authorization flow related factories
+ * @internal
+ */
+trait AuthFlowFactoryTrait
```

Comment:
I don't think we need this Trait. A simple `$container->get(AuthorizationServer::class)` is enough in each factory class. All the checks present in `AuthFlowFactoryTrait` are performed by the service manager by default.

Reply:
Agree. Getting rid of the additional checks will simplify the codebase and increase maintainability.

I just ported it from the existing `OAuth2MiddlewareFactory`, because I didn't know if there was a specific intend for the specialized exception.